gse-afs-ext-base/src/main/java/com/powsybl/gse/afs/ext/base/ImportedCaseCreator.java

```diff
@@ -163,4 +176,46 @@ public BooleanBinding okProperty() {
     @Override
     public void dispose() {
     }
+
+    private void replaceNode(ProjectFolder folder, Case aCase, Scene scene) {
+        String name = aCase.getName();
+        Optional<ButtonType> result = GseAlerts.showReplaceAndQuitDialog(folder.getName(), name);
+        result.ifPresent(buttonType -> {
+            if (buttonType.getButtonData() == ButtonBar.ButtonData.YES) {
+                folder.getChild(name).ifPresent(projectNode -> {
+                    backwardDependencies = projectNode.getBackwardDependencies();
+                    projectNode.rename(TEMPORARY_NAME + UUID.randomUUID());
+                    buildFile(aCase, folder, null);
+                    Optional<ProjectNode> importedNode = folder.getChild(name);
+                    if (importedNode.isPresent()) {
+                        backwardDependencies.forEach(projectFile -> projectFile.replaceDependency(projectNode.getId(), importedNode.get()));
+                        projectNode.delete();
+                    } else {
+                        projectNode.rename(name);
+                    }
+                });
+            } else if (buttonType.getButtonData() == ButtonBar.ButtonData.OTHER) {
+                folder.getChild(name).ifPresent(projectNode -> {
+                    Optional<String> text = RenamePane.showAndWaitDialog(scene.getWindow(), projectNode);
+                    text.ifPresent(newName -> {
+                        if (!folder.getChild(newName).isPresent()) {
+                            buildFile(aCase, folder, newName);
+                        } else {
+                            replaceNode(folder, aCase, scene);
```

Comment:
here in this case you will try to replace/rename the node using the original name, not the new one

Reply:
it's actually a new name. the file is build using a different name to the original one

Reply:
i'm talking about the recursion with replaceNode

Reply:
it's true. and the recursion method (replaceNode) is not necessary because the rename pane prevent this case

Reply:
no, the replaceNode recursion is necessary, or at least recurse until the renaming is done without conflict